usaspending_api/awards/v2/filters/location_filter_geocode.py

```diff
@@ -1,82 +1,124 @@
+            if v.get('county'):
+                nested_locations[v['country']][v['state']]['county'].extend(get_fields_list('county', v['county']))
 
-        if type(model) == str:
-            model = apps.get_model(default_model, model)
-        qs = model.objects.filter(**kwargs)
+            if v.get('district'):
+                nested_locations[v['country']][v['state']]['district'].extend(
+                    get_fields_list('district', v['district']))
 
-        if queryset_init:
-            or_queryset |= qs
-        else:
-            queryset_init = True
-            or_queryset = qs
-    return or_queryset
+        except KeyError:
+            # This result if there is value for country or state where necessary
+            location_error_handling(v.keys())
 
+    return nested_locations
 
-def check_location_fields(fields):
+
+def location_error_handling(fields):
     # Request must have country, and can only have 3 fields,
     # and must have state if there is county or district
-    if 'country' not in fields or \
-            ('state' not in fields and
-                ('county' in fields or 'district' in fields)):
+    if 'country' not in fields:
 
         raise InvalidParameterException(
-            'Invalid filter: recipient has incorrect object.'
+            'Invalid filter:  Missing necessary location field: country.'
+        )
+    elif 'state' not in fields and('county' in fields or 'district' in fields):
+        raise InvalidParameterException(
+            'Invalid filter:  Missing necessary location field: state.'
         )
 
 
-def get_fields_list(scope, field_value, loc_dict):
+def get_fields_list(scope, field_value):
     """List of values to search for; `field_value`, plus possibly variants on it"""
     if scope in ['congressional_code', 'county_code']:
         try:
+            # Congressional and county codes are not uniform and contain multiple variables
+            # In the location table Ex congressional code (01): '01', '1.0', '1'
             return [str(int(field_value)), field_value, str(float(field_value))]
         except ValueError:
             # if filter causes an error when casting to a float or integer
             # Example: 'ZZ' for an area without a congressional code
-            return [field_value]
+            pass
     return [field_value]
 
 
-def return_query_strings(use_matview):
+def return_query_string(use_matview):
     # Returns query strings according based on mat view or database
-    loc_dict = LOCATION_MAPPING.copy()
-    q_str = '{0}__{1}__in'
 
     if use_matview:
-        q_str = '{0}_{1}__in'
-        loc_dict['country'] = 'country_code'
-
-    return q_str, loc_dict
+        # Queries going through the references_location table will not require a join
+        # Example "pop__county_code"
+        q_str = '{0}_{1}'
+        # Matviews use country_code ex: pop_country_code
+        country_code = 'country_code'
```

Comment:
can this variable be named something more indicative of the value it's storing? Something along the lines of `country_code_column`?